### DISCUSSION TITLE
[.NET 8 COM support] CefSharp 120.1.110

### DISCUSSION THREAD

**Discussion by pieseczek** [cefsharp/CefSharp#4672] [General]
Hi,
I would like to use CefSharp with the latest .NET 8. 
I am using COM in my project. COM object is the client of CefSharp library, I am using CefSharp library in the constructor of the Foo class.
When I am creating the COM object, in the Foo constructor the exception is thrown:
_
> System.IO.FileNotFoundException: 'Could not load file or assembly 'CefSharp.Core.Runtime, Version=120.1.110.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138'. The specified module could not be found.'
> FileNotFoundException: Could not load file or assembly '<ProjectWithFooClassPath>\bin\x86\Debug\net8.0-windows\runtimes\win-x86\lib\netcoreapp3.1\CefSharp.Core.Runtime.dll'. The specified module could not be found.
_

I have run the same code with .NET 7 -> the same exception was raised.
The code is working successfully with .net48.

```
[assembly: ComVisible(true)]
namespace Foo
{
    [Guid("164ba864-95a5-4c15-8d72-8f6a325ce75c"),
    ComVisible(true),
    ClassInterface(ClassInterfaceType.AutoDispatch)]
    public class Foo
    {
        public Foo()
        {
            var settings = new CefSettings();
        }
    }
}

<PackageReference Include="CefSharp.WinForms.NETCore" Version="120.1.110" />
<PackageReference Include="CefSharp.Common.NETCore" Version="120.1.110" />
```

The COM object creation:
```
        var clsid = "{164ba864-95a5-4c15-8d72-8f6a325ce75c}";
        var type = Type.GetTypeFromCLSID(new Guid(clsid));
        var obj = Activator.CreateInstance(type, true);
```

Before running I have registered *.com.dll and verified that the GUID was added correctly in system registry,

Could you help me with these issue? Maybe some workaround could be applied?

**Comment by amaitland** [comment]
I've never tried using `CefSharp` via `COM`, so the only advice I can give you is generic.

Make sure you've set a `RuntimeIdentifier` so you have an arch specific build.

- https://github.com/cefsharp/CefSharp/blob/master/NuGet/PackageReference/Readme.txt#L10
- https://learn.microsoft.com/en-us/dotnet/core/rid-catalog#windows-rids
